server/src/main/java/org/elasticsearch/cluster/node/DiscoveryNodes.java

```diff
@@ -478,16 +482,25 @@ public boolean added() {
         public String shortSummary() {
             StringBuilder sb = new StringBuilder();
             if (!removed() && masterNodeChanged()) {
-                if (newMasterNode.getId().equals(localNodeId)) {
-                    // we are the master, no nodes we removed, we are actually the first master
-                    sb.append("new_master ").append(newMasterNode());
+                if (newMasterNode() != null) {
```

Comment:
do we need all this string crafting? can we just check for change and say "its was x, now it's y"?

Reply:
That would simplify the summary, indeed

Reply:
Yes, there is some effort involved with string crafting. But I think it helps the end user understand what happened, for example,
- "stepped down as master" (if localNode == previousMaster && newMaster == null)
- "became master" (if localNode == newMaster)
- "lost master" (if localNode != previousMaster && newMaster == null)
- "detected master" (newMaster != null && newMaster != localNode)
- ...?

Reply:
I'm good if you guys think it's worth it. My only concern is that people will start thinking stuff that are necessarily true - (or might be not true in the future) - like are we guaranteed to always have an even with a master stepped down? Maybe we have a master swap, or a restart and the intermediate cluster state was lost in batching?

Reply:
We talked with @ywelsch and we agreed on logging a simple message with previous/current master values for now, as crafting more detailed messages could lead the user to expect a specific message while the related event could be lost because of cluster state batching etc. 

For now it will log messages like:
```
[node_tm0] master node changed {previous [], current [{node_tm1}{...}
[node_tm2] master node changed {previous [{node_tm2}{...}], current []}, reason: not enough master nodes
``` 
but we may revisit this in the future.